imap/guesstz.h

```diff
@@ -0,0 +1,66 @@
+/* guesstz.h -- routines to guess timezone ids from VTIMEZONEs
+ *
+ * Copyright (c) 1994-2021 Carnegie Mellon University.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *
+ * 3. The name "Carnegie Mellon University" must not be used to
+ *    endorse or promote products derived from this software without
+ *    prior written permission. For permission or any legal
+ *    details, please contact
+ *      Carnegie Mellon University
+ *      Center for Technology Transfer and Enterprise Creation
+ *      4615 Forbes Avenue
+ *      Suite 302
+ *      Pittsburgh, PA  15213
+ *      (412) 268-7393, fax: (412) 268-7395
+ *      innovation@andrew.cmu.edu
+ *
+ * 4. Redistributions of any form whatsoever must retain the following
+ *    acknowledgment:
+ *    "This product includes software developed by Computing Services
+ *     at Carnegie Mellon University (http://www.cmu.edu/computing/)."
+ *
+ * CARNEGIE MELLON UNIVERSITY DISCLAIMS ALL WARRANTIES WITH REGARD TO
+ * THIS SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS, IN NO EVENT SHALL CARNEGIE MELLON UNIVERSITY BE LIABLE
+ * FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+ * AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
+ * OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#ifndef GUESSTZ_H
+#define GUESSTZ_H
+
+#include <jansson.h>
+
+#include "ical_support.h"
+
+struct guesstzdb;
+
+extern struct guesstzdb *guesstz_open(const char *alt_fname);
+extern void guesstz_close(struct guesstzdb **gtzdbptr);
+
+extern int guesstz_create(const char *zoneinfo_dir, const char *fname,
+                            icaltimetype start, icaltimetype end);
+
+extern char *guesstz_dump(const char *fname);
+
+extern void guesstz_toiana(struct guesstzdb *gtzdb,
+                           struct buf *idbuf, icalcomponent *vtz,
+                           struct icalperiodtype span,
+                           unsigned is_recurring);
+
```

Comment:
It might be helpful here or in .c to have a comment describing the three different record types and then 'O' records point to 'T' records (and any other subtleties).

Reply:
I added both a comment describing the database schema, as well as documented the exported functions.